views/errors.go

```diff
@@ -2,13 +2,23 @@ package views
 
 import (
 	"github.com/labstack/echo/v4"
+	"github.com/labstack/gommon/log"
 	"github.com/ystv/web-auth/templates"
 )
 
-func (v *Views) Error404(c echo.Context) error {
-	return v.template.RenderNoNavsTemplate(c.Response().Writer, nil, templates.NotFound404Template)
+func (v *Views) CustomHTTPErrorHandler(err error, c echo.Context) {
+	log.Print(err)
+	data := struct {
```

Comment:
You can check the error code here with something like this (note: untested):
```suggestion
	var statusCode int
	if he, ok := error.(*echo.HTTPError); ok {
		statusCode = he.Code
	} else {
		statusCode = 500
	}
	// do something with statusCode
	
	data := struct {
```

Reply:
I've tried it but still doesn't work, reason being is because of the custom error handle func, nothing is returned so no changes will persist past the function

Reply:
Never mind, found something different that works, c.Response().WriteHeader(status)

Reply:
Will commit what I have